jax/lax/lax_control_flow.py

```diff
@@ -850,6 +850,29 @@ def body(i, dst):
 masking.masking_rules[lax.concatenate_p] = _concat_masking_rule
 
 
+def _flatten_high_level_func(
+    fun, tree, error_template="Expected {}, got {}",
+):
+  """Flatten a higher level function ``f`` of the form ``f(g, x)``.
```

Comment:
We'd usually use "higher order" rather than "high level" to describe a function that takes a function as an argument.